catalog/spatial/registry/registry-ebrim-transformer/src/test/java/org/codice/ddf/registry/transformer/RegistryTransformerTest.java

```diff
@@ -118,26 +118,52 @@ public void testBasicInfo() throws Exception {
         MetacardImpl meta = convert("/csw-basic-info.xml");
         assertRegistryMetacard(meta);
 
-        assertThat(meta.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
-                .getValue(), is("urn:uuid:2014ca7f59ac46f495e32b4a67a51276"));
+        assertThat(RegistryUtility.getStringAttribute(meta,
+                RegistryObjectMetacardType.REGISTRY_ID,
+                null), is("urn:uuid:2014ca7f59ac46f495e32b4a67a51276"));
         assertThat(meta.getTitle(), is("my service"));
         assertThat(meta.getDescription(), is("something"));
         assertThat(meta.getContentTypeVersion(), is("0.0.0"));
     }
 
     @Test
     public void testOrgInfo() throws Exception {
-        MetacardImpl meta = convert("/csw-org-info.xml");
-        assertRegistryMetacard(meta);
+        MetacardImpl metacard = convert("/csw-org-info.xml");
+        assertRegistryMetacard(metacard);
+
+        assertThat(RegistryUtility.getStringAttribute(metacard,
+                RegistryObjectMetacardType.ORGANIZATION_NAME,
+                null), is("Codice"));
+        assertThat(RegistryUtility.getListOfStringAttribute(metacard,
+                RegistryObjectMetacardType.ORGANIZATION_ADDRESS),
+                hasItem("1234 Some Street, Phoenix, AZ 85037, USA"));
+        assertThat(RegistryUtility.getListOfStringAttribute(metacard,
+                RegistryObjectMetacardType.ORGANIZATION_PHONE_NUMBER),
+                hasItem("(555) 555-5555 extension 1234"));
+        assertThat(RegistryUtility.getListOfStringAttribute(metacard,
+                RegistryObjectMetacardType.ORGANIZATION_EMAIL),
+                hasItem("emailaddress@something.com"));
+    }
 
-        assertThat(meta.getAttribute(RegistryObjectMetacardType.ORGANIZATION_NAME)
-                .getValue(), is("Codice"));
-        assertThat(meta.getAttribute(RegistryObjectMetacardType.ORGANIZATION_ADDRESS)
-                .getValue(), is("1234 Some Street, Phoenix, AZ 85037, USA"));
-        assertThat(meta.getAttribute(RegistryObjectMetacardType.ORGANIZATION_PHONE_NUMBER)
-                .getValue(), is("(555) 555-5555 extension 1234"));
-        assertThat(meta.getAttribute(RegistryObjectMetacardType.ORGANIZATION_EMAIL)
-                .getValue(), is("emailaddress@something.com"));
+    @Test
+    public void testNullMetacardAttributeValues() throws Exception {
+        MetacardImpl metacard = convert("/csw-null-metacard-attributes.xml");
+        assertRegistryMetacard(metacard);
+
+        assertThat(RegistryUtility.getStringAttribute(metacard,
+                RegistryObjectMetacardType.ORGANIZATION_NAME,
+                null), is("Codice"));
+        assertThat(RegistryUtility.getListOfStringAttribute(metacard,
```

Comment:
We're not testing any of the scenarios where a piece of the address is `null` or missing. We're also not testing many of the other scenarios such as what if the extension isn't there.

Maybe we should extract `StreetAddressStringBuilder` and `PhoneNumberStringBuilder` classes and unit test those independently. That way you wouldn't have to do that low-level testing here.